app/process/handler/consensus-message/consensus-message.go

```diff
@@ -84,6 +93,34 @@ func (cmh ConsensusMessageHandler) Handle(payload []byte) {
 func (cmh ConsensusMessageHandler) handleEthTxMessage(m *validatorproto.TopicEthTransactionMessage) error {
 	// TODO: verify authenticity of transaction hash
 
+	err := cmh.transactionRepository.UpdateStatusEthTxSubmitted(m.TransactionId, m.EthTxHash)
+	if err != nil {
+		cmh.logger.Errorf("Failed to update status to [%s] of transaction with TransactionID [%s]. Error [%s].", transaction.StatusEthTxSubmitted, m.TransactionId, err)
+		return err
+	}
+
+	isSuccessful, err := cmh.ethereumClient.WaitForTransactionSuccess(common.HexToHash(m.EthTxHash))
+	if err != nil {
+		cmh.logger.Errorf("Failed to await TX ID [%s] with ETH TX [%s] to be mined. Error [%s].", m.TransactionId, m.Hash, err)
+		return err
+	}
+
+	if !isSuccessful {
+		cmh.logger.Infof("Transaction with ID [%s] was reverted. Updating status to [%s].", m.TransactionId, transaction.StatusEthTxReverted)
+		err = cmh.transactionRepository.UpdateStatusEthTxReverted(m.TransactionId)
+		if err != nil {
+			cmh.logger.Errorf("Failed to update status to [%s] of transaction with TransactionID [%s]. Error [%s].", transaction.StatusEthTxReverted, m.TransactionId, err)
+			return err
+		}
+	} else {
+		cmh.logger.Infof("Transaction with ID [%s] was successfully mined. Updating status to [%s].", m.TransactionId, transaction.StatusCompleted)
+		err = cmh.transactionRepository.UpdateStatusCompleted(m.TransactionId)
+		if err != nil {
+			cmh.logger.Errorf("Failed to update status to [%s] of transaction with TransactionID [%s]. Error [%s].", transaction.StatusCompleted, m.TransactionId, err)
+			return err
+		}
+	}
```

Comment:
I think that this process is a blocking one. If new ethereum tx is broadcasted we wouldn't want to block at line `102` to wait for the TX to be mined. We would ideally:
1. Update that the TX was submitted
2. Cancel the scheduled submission in the `scheduler`
3. Run a non-blocking (go routine) to wait for tx mining and update the status afterwards (completed/reverted)

At the moment, we are blocking the waiting and canceling the scheduler only after the TX is mined .
For reference you can check the invocation of `checkEthTransactionStatus` in crypto-transfer handler